### PR TITLE
Enable debugging of asynchronous tasks in Intellij (#12316)

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:
Evented systems like Netty can be much harder to debug than sequential and blocking code.
Fortunately, with a few strategically placed hints, we can significantly improve the tooling support.

Modification:
Add a dependency on the JetBrains annotations, and use the `Schedule` and `Execute` annotations to tell the Intellij IDEA debugger how our event loops and executors handle asynchronous task execution.
Basically, we tell it where tasks are scheduled for asynchronous execution, and we tell it where this execution is picked up.

Result:
The Intellij IDEA debugger can then bridge the schedule and execute points, and present a unified, or spliced stack trace.
This way, we are able to see who scheduled what tasks, and why.

### CODE REVIEW

**Review by franz1981 — APPROVED** [review]
I saw it while pushed by Chris...and didn't know about this feature...it's super cool

File: pom.xml [hunk 1]
```diff
@@ -786,6 +786,14 @@
         <version>3.1.0</version>
       </dependency>
 
+      <!-- Annotations for IDE integration and analysis -->
+      <dependency>
+        <groupId>org.jetbrains</groupId>
+        <artifactId>annotations</artifactId>
```
Comment (chrisvest) [inline pom.xml:792]: Since 4.1 has Java 6 compatibility, I think we need to use the `annotations-java5` variant. The one pulled in here expects Java 8 or better.

**Review by chrisvest — APPROVED** [review]
